codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -526,32 +528,29 @@ private boolean exceedsMaxHeaderSize(long size) {
         return true;
     }
 
-    private byte[] readStringLiteral(InputStream in, int length) throws IOException {
-        byte[] buf = new byte[length];
-        if (in.read(buf) != length) {
-            throw DECOMPRESSION_EXCEPTION;
-        }
-
+    private CharSequence readStringLiteral(ByteBuf in, int length) throws IOException {
         if (huffmanEncoded) {
-            return Huffman.DECODER.decode(buf);
+            return huffmanDecoder.decode(in, length);
         } else {
-            return buf;
+            byte[] buf = new byte[length];
+            in.readBytes(buf);
```

Comment:
@Scottmitch I think this will not buy us anything as then we will need to make the copy in AsciiString.